### PR TITLE
Updates:

### PR DESCRIPTION
**Pull request by grandizzy**
- add Makefile
- add GH actions run tests with gas report and coverage step
- Ajna token:
  - update Solidity to 0.8.15
  - add guard against sending tokens to token contract
  - initialize in token contract constructor
- Ajna token tests

### CODE REVIEW

**Review by EdNoepel — APPROVED** [review]
LGTM

File: src/Token.sol [hunk 1]
```diff
@@ -8,11 +8,6 @@ import { ERC20PermitUpgradeable } from "@openzeppelin/openzeppelin-contracts-upg
 import { ERC20VotesUpgradeable } from "@openzeppelin/openzeppelin-contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
 
 contract AjnaToken is Initializable, ERC20Upgradeable, ERC20BurnableUpgradeable, ERC20PermitUpgradeable, ERC20VotesUpgradeable {
-    /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor() {
-        initialize();
-        _disableInitializers();
-    }
```
Comment (MikeHathaway) [inline src/Token.sol:14]: I think it's still recommended to call `_disableInitializers()` in the constructor() without the `initialize()` there

**Comment by MikeHathaway** [comment]
Updated my PR with the fix for interacting with the upgradeable token in the forge tests here: https://github.com/ajna-finance/ajna-token/pull/1

**Review by MikeHathaway — CHANGES_REQUESTED** [review]
Tests look good, but this should be rebased on the other PR following merge

**Comment by grandizzy** [comment]
> Tests look good, but this should be rebased on the other PR following merge

I created PR https://github.com/ajna-finance/ajna-token/pull/3 on top of `token-tests` branch to add only missing things from this PR

**Pull request closed without merging**